metricflow/test/model/validations/test_data_warehouse_tasks.py

```diff
@@ -1,52 +1,99 @@
+from copy import deepcopy
+import pytest
+from metricflow.engine.metricflow_engine import MetricFlowEngine
 from metricflow.model.data_warehouse_model_validator import (
     DataWarehouseModelValidator,
     DataWarehouseTaskBuilder,
     DataWarehouseValidationTask,
 )
+from metricflow.model.model_transformer import ModelTransformer
 
 from metricflow.model.objects.data_source import DataSource, Mutability, MutabilityType
-from metricflow.model.objects.elements.dimension import Dimension, DimensionType
+from metricflow.model.objects.elements.dimension import Dimension, DimensionType, DimensionTypeParams
+from metricflow.model.objects.elements.measure import AggregationType, Measure
 from metricflow.model.objects.user_configured_model import UserConfiguredModel
+from metricflow.model.semantic_model import SemanticModel
 from metricflow.model.validations.validator_helpers import ValidationIssueLevel
+from metricflow.plan_conversion.column_resolver import DefaultColumnAssociationResolver
+from metricflow.plan_conversion.time_spine import TimeSpineSource
 from metricflow.protocols.sql_client import SqlClient
-from metricflow.specs import DimensionReference
+from metricflow.specs import DimensionReference, MeasureReference
+from metricflow.test.fixtures.setup_fixtures import MetricFlowTestSessionState
+from metricflow.test.test_utils import as_datetime
+from metricflow.test.time.configurable_time_source import ConfigurableTimeSource
+from metricflow.time.time_granularity import TimeGranularity
 
 
-def test_build_data_source_tasks() -> None:  # noqa:D
-    dim_ref_1 = DimensionReference(element_name="dim1")
-    dim_ref_2 = DimensionReference(element_name="dim1")
-
+@pytest.fixture
+def basic_model() -> UserConfiguredModel:  # noqa: D
```

Comment:
We've been using YAML files to define models - can you take a look at `metricflow/test/fixtures/model_yamls`?

Reply:
Also `metricflow/test/fixtures/model_fixtures.py` would be relevant as well.

Reply:
In an iteration of this testing file we had been using the `simple_semantic_model` in `model_fixtures.py`. I had reasons for refactoring away from it in this test suite, but I now forget what those were 🙃 In the commit where this happened I added the following as part of the commit message

> Additionally, these tests need an easy to use model which is always valid by default. Thus an entirely self referential model was added as a fixture.

It may be that the `simple_semantic_model` wasn't succeeding the validations we were checking in these tests for whatever reason. I'll double check to see if that's the case.

Reply:
Yea so I tried swapping out the `basic_model` I defined as a fixture for the `simple_user_configured_model` fixture defined in `model_fixtures.py` and without any changes that model has 9 data source validation issues and hits `Unkown element 'lux_or_normal_listing'` when generating metric queries. I believe the data source issues are because the data warehouse queried during testing is empty or has a different schema. The reason I built out the `basic_model` fixture was to make an entirely self referential model, i.e. it works against even an empty data warehouse.

Reply:
Paul and I chatted offline. I misread Paul's comments 🤦 We're going to formalize this model so that it's available to other tests as well as a fixture.